### ISSUE TITLE
Connected accounts are not listed.

### ISSUE THREAD

**Issue by mooorex** [ethereum/remix-project#5432]
**Describe the bug**
On the tab of Deploy & run transactions, when the environment is switched from others (e.g. Remix VM(Cancun), the default environment) to injected provider-metamask, if the injected provider is already connected, the account will not be listed.

**To Reproduce**
1. Go to the tab of Deploy & run transactions
2. Click on the Environment drop-down box and select Injected Provider - Metamask
3. Connect Remix to the Injected provider - Metamask and see the account drop-down box is filled with connected accounts
4. Click on the Environment drop-down box and select Remix VM(Cancun)
5. Click on the Environment drop-down box and select Injected Provider - Metamask
6. See error, the account drop-down box is blank

**Expected behavior**
the account drop-down box is filled with connected accounts.

**Screenshots**
![problem-screenshot](https://github.com/user-attachments/assets/064f814b-764a-43f0-8b22-62c9f73f2073)

**Desktop (please complete the following information):**
 - OS: MacOS 15.1
 - Browser: Chrome 131.0.6778.86 (Official Build) (arm64)
 - Metamask 12.7.2

**Additional context**
The problem does not exist on my Arch Linux operating system (kernel 6.12.1-arch1-1) equipped with chromium-131.0.6778.85-1.

**Comment by Aniket-Engg** [comment]
@mooorex I cannot reproduce the issue with shared 'steps to reproduce'.

Do you see it each time and every network? I mean is it a consistent issue?

**Comment by Aniket-Engg** [comment]
Sure. Thanks for the info 👍

**Comment by mooorex** [comment]
Besides metamask, I installed another extension wallet, then the account list is empty.  

If I turn off this "another extension wallet" in the page chrome://extensions/, the Remix works very well.

Then, the second extension wallet causes the trouble. @Aniket-Engg

**Comment by LianaHus** [comment]
are you you you are allowing connection from wallet side?

**Comment by mooorex** [comment]
Yes, as you see in the screenshot, Metamask already shows connected. @LianaHus

**Comment by lewinskimaciej** [comment]
Can confirm, account selection and contract deploying doesn't work if you have more than one wallet installed(in my case MetaMask and Ava Core).

**Comment by ritvikPuranik** [comment]
The issue looks like when there's no account selected explicitly(for injected providers), the code is erroring out. I have added a fix to fallback to the first account if no account is explicitly selected. in my commit above, you can see the issue and the result after the fix. let me know if it persists

**Comment by ChengHua926** [comment]
As of 3/6/2025, I ran into that same problem again. I had Phantom and Metamask installed, and the address was not listed until I turned off Phantom

**Comment by princ3szn** [comment]
3/15/2025, got the same issue. i had multiple wallets installed including metamask but my wallet was not reflecting till i turned them all of.

**Comment by joeizang** [comment]
This issue has since been fixed. Please make sure your browser is updated and your browser refreshed.

**Comment by mooorex** [comment]
> This issue has since been fixed. Please make sure your browser is updated and your browser refreshed.

Confirmed. It works well now, thanks!

**Comment by joeizang** [comment]
This has been resolved. I am closing this.
